flyteadmin/scheduler/core/gocron_scheduler_test.go

```diff
@@ -197,11 +197,126 @@ func TestGetTimedFuncWithSchedule(t *testing.T) {
 }
 
 func TestGetCronScheduledTime(t *testing.T) {
-	fromTime := time.Date(2022, time.January, 27, 19, 0, 0, 0, time.UTC)
-	nextTime, err := getCronScheduledTime("0 19 * * *", fromTime)
-	assert.Nil(t, err)
-	expectedNextTime := time.Date(2022, time.January, 28, 19, 0, 0, 0, time.UTC)
-	assert.Equal(t, expectedNextTime, nextTime)
+	tests := []struct {
+		name           string
+		cronExpression string
+		fromTime       time.Time
+		expectError    bool
+		errorContains  string
+		expectedTime   time.Time
+	}{
+		{
+			name:           "valid cron expression",
+			cronExpression: "0 19 * * *",
+			fromTime:       time.Date(2022, time.January, 27, 19, 0, 0, 0, time.UTC),
+			expectError:    false,
+			expectedTime:   time.Date(2022, time.January, 28, 19, 0, 0, 0, time.UTC),
+		},
+		{
+			name:           "invalid cron expression with February 31st should return error",
+			cronExpression: "0 0 31 2 *", // February 31st - invalid
+			fromTime:       time.Date(2023, time.January, 1, 0, 0, 0, 0, time.UTC),
+			expectError:    true,
+			errorContains:  "invalid crontab configuration",
+		},
+		{
+			name:           "invalid cron expression with April 31st should return error",
+			cronExpression: "0 0 31 4 *", // April 31st - invalid
+			fromTime:       time.Date(2023, time.January, 1, 0, 0, 0, 0, time.UTC),
+			expectError:    true,
+			errorContains:  "invalid crontab configuration",
+		},
+		{
+			name:           "invalid cron expression with malformed syntax should return error",
+			cronExpression: "invalid cron expression",
+			fromTime:       time.Date(2023, time.January, 1, 0, 0, 0, 0, time.UTC),
+			expectError:    true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			nextTime, err := getCronScheduledTime(tt.cronExpression, tt.fromTime)
+
+			if tt.expectError {
+				assert.NotNil(t, err)
+				if tt.errorContains != "" {
+					assert.Contains(t, err.Error(), tt.errorContains)
+				}
+				assert.True(t, nextTime.IsZero())
+				t.Logf("Got expected error: %v", err)
+			} else {
+				assert.Nil(t, err)
+				assert.Equal(t, tt.expectedTime, nextTime)
+			}
+		})
+	}
+}
+
+func TestGetCatchUpTimesWithInvalidCronExpression(t *testing.T) {
```

Comment:
```suggestion
func TestGetCatchUpTimesWithCronExpression(t *testing.T) {
```

We also got a valid test case